src/Formula.elm

```diff
@@ -163,9 +189,49 @@ binary conn constructor =
   |. spaces
   |. symbol ")"
 
+args : Parser (List Term)
+args =
+  succeed (::)
+    |. symbol "("
+    |. spaces
+    |= lazy (\_ -> term)
+    |. spaces
+    |= lazy (\_ -> repeat zeroOrMore nextArg)
+    |. symbol ")"
+
+nextArg : Parser Term
+nextArg =
+  succeed identity
+    |. symbol ","
+    |. spaces
+    |= term
+    |. spaces
+
+term : Parser Term
+term =
+  variable isLetter isIdentChar Set.empty |>
```

Comment:
Function symbols, predicate symbols, and variables are distinguished only by their use, not by any lexical restrictions.

I imagine that the editor will restrict improper use of symbols in a tableau by automatically collecting the three sets of symbols, and reporting an error if they are not disjoint.